src/UNL/Peoplefinder/Developers/AbstractResource.php

```diff
@@ -0,0 +1,53 @@
+<?php
+abstract class UNL_PeopleFinder_Developers_AbstractResource
+{
+    const FORMAT_JSON = 'json';
+    const FORMAT_XML = 'xml';
+    const FORMAT_PARTIAL = 'partial';
+    
+    /**
+     * @return string - a brief description of the resource
+     */
+    abstract public function getTitle();
+    
+    /**
+     * @return string - a brief description of the resource
+     */
+    abstract public function getDescription();
+
+    /**
+     * @return mixed - an associative array of property=>description
+     */
+    abstract public function getAvailableFormats();
+
+    /**
+     * @return array - an associative array of property=>description
+     */
+    abstract public function getJsonProperties();
+
+    /**
+     * @return array - an associative array of property=>description
+     */
+    abstract public function getXmlProperties();
```

Comment:
I agree 100%. That being said, they don't return the same data. And there will always be the chance for slight differences between formats which should be documented independently. IF the data is in fact the same (which should be the goal), you could just return the results from another method instead of defining them again.